File: backend/src/test/java/ca/metricalsky/winston/entity/ThumbnailEntityTest.java
package ca.metricalsky.winston.entity;

import ca.metricalsky.winston.test.TestUtils;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
class ThumbnailEntityTest {

    @Autowired
    private TestEntityManager entityManager;

    @Test
    void persists() {
        var thumbnailEntity = ThumbnailEntity.builder()
                .id(TestUtils.randomId())
                .url(TestUtils.randomString())
                .image(TestUtils.randomBytes(1000))
                .build();

        var persistedEntity = entityManager.persistFlushFind(thumbnailEntity);

        assertThat(persistedEntity)
                .hasFieldOrPropertyWithValue("id", thumbnailEntity.getId())
                .hasFieldOrPropertyWithValue("url", thumbnailEntity.getUrl())
                .hasFieldOrPropertyWithValue("image", thumbnailEntity.getImage())
                .hasNoNullFieldsOrProperties();
    }
}

File: database/src/test/kotlin/ca/metricalsky/winston/database/entity/ThumbnailEntityTest.kt
package ca.metricalsky.winston.database.entity

import ca.metricalsky.winston.database.test.annotation.DatabaseTest
import ca.metricalsky.winston.database.test.faker.DatabaseFaker
import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import io.kotest.matchers.types.shouldNotBeSameInstanceAs
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager

@DatabaseTest
class ThumbnailEntityTest(
    entityManager: TestEntityManager,
): WordSpec({

    val faker = DatabaseFaker()

    "ThumbnailEntity" should {

        "persist" {
            val thumbnail = faker.thumbnail().completeEntity()

            entityManager.persistFlushFind(thumbnail) shouldNotBeNull {
                this shouldNotBeSameInstanceAs thumbnail
                id shouldBe thumbnail.id
                url shouldBe thumbnail.url
                image shouldBe thumbnail.image
            }
        }
    }
})
